[databricks] `LAG` function crashes the parser

**Fully reproducible code snippet**

```python
def test_y():
    query = """
select
  all.XXX,
  all.YYY,
  lag(all.XXX) over (
    partition by all.ZZZ,
    all.WWW,
    all.KKK
    order by
      all.XXX
  ) prev_XXX
from values
(1,2,3,4,5),
(6,7,8,9,10),
(1,2,3,4,5)
as all(XXX, YYY, ZZZ, WWW, KKK)
"""
    x = sqlglot.parse(query, dialect='databricks')
    assert x
```

produces

```
E           sqlglot.errors.ParseError: Invalid expression / Unexpected token. Line 3, Col: 5.
E             
E           select
E             all.XXX,
E             all.YYY,
E             lag(all.XXX) over (
E               partition by all.ZZZ,
E               all.WWW,
E               all.KKK
E               order b
```

but is a valid query:

<img width="379" alt="image" src="https://github.com/user-attachments/assets/531705eb-f49f-4780-a677-1fcfaa7d56d3">

Blocks:
- https://github.com/databrickslabs/ucx/issues/2235

**Official Documentation**
https://docs.databricks.com/en/sql/language-manual/functions/lag.html

--- a/sqlglot/parser.py
+++ b/sqlglot/parser.py
@@ -376,6 +376,7 @@ class Parser(metaclass=_Parser):
 
     # Tokens that can represent identifiers
     ID_VAR_TOKENS = {
+        TokenType.ALL,
         TokenType.VAR,
         TokenType.ANTI,
         TokenType.APPLY,
@@ -2742,8 +2743,12 @@ class Parser(metaclass=_Parser):
             comments = self._prev_comments
 
             hint = self._parse_hint()
-            all_ = self._match(TokenType.ALL)
-            distinct = self._match_set(self.DISTINCT_TOKENS)
+
+            if self._next and not self._next.token_type == TokenType.DOT:
+                all_ = self._match(TokenType.ALL)
+                distinct = self._match_set(self.DISTINCT_TOKENS)
+            else:
+                all_, distinct = False, False
 
             kind = (
                 self._match(TokenType.ALIAS)

--- a/tests/fixtures/identity.sql
+++ b/tests/fixtures/identity.sql
@@ -871,3 +871,4 @@ SELECT unnest
 SELECT * FROM a STRAIGHT_JOIN b
 SELECT COUNT(DISTINCT "foo bar") FROM (SELECT 1 AS "foo bar") AS t
 SELECT vector
+WITH all AS (SELECT 1 AS count) SELECT all.count FROM all

